src/client/shader.cpp

```diff
@@ -620,15 +653,58 @@ ShaderInfo generate_shader(const std::string &name, u8 material_type, u8 drawtyp
 		return shaderinfo;
 
 	// Create shaders header
-	std::string shaders_header = "#version 120\n";
-
+	bool use_gles = false;
+#if ENABLE_GLES
+	use_gles = driver->getDriverType() == video::EDT_OGLES2;
+#endif
+	std::string shaders_header, vertex_header, pixel_header; // geometry shaders aren’t supported in GLES<3
+	if (use_gles) {
+		shaders_header = \
+			"#version 100\n" \
+			"varying mediump vec4 varTexCoord[1];\n" \
+			"#define gl_TexCoord varTexCoord\n" \
+			;
+		vertex_header = \
+				"uniform highp mat4 mWorldView;\n" \
+				"uniform mat4 mTexture[1];\n" \
+				"uniform mat3 mNormal;\n" \
+				"attribute highp vec4 inVertexPosition;\n" \
+				"attribute vec3 inVertexNormal;\n" \
+				"attribute vec4 inVertexColor;\n" \
+				"attribute vec4 inTexCoord0;\n" \
+	/* 			"attribute vec4 inTexCoord1;\n" */\
+				"attribute vec4 inVertexTangent;\n" \
+				"attribute vec4 inVertexBinormal;\n" \
+				"\n" \
+				"#define gl_ModelViewMatrix mWorldView\n" \
+				"#define gl_TextureMatrix mTexture\n" \
+				"#define gl_NormalMatrix mNormal\n" \
+				"#define gl_Vertex inVertexPosition\n" \
+				"#define gl_Normal inVertexNormal\n" \
+				"#define gl_Color inVertexColor\n" \
+				"#define gl_MultiTexCoord0 inTexCoord0\n" \
+				"#define gl_MultiTexCoord1 inVertexTangent\n" \
+				"#define gl_MultiTexCoord2 inVertexBinormal\n" \
+				;
+		pixel_header = \
+			"precision mediump float;\n" \
+			;
+	} else {
+		shaders_header = \
+			"#version 120\n" \
+			;
+	}
+
+
+	bool use_discard = use_gles;
```

Comment:
why is this turned on by default?

Reply:
because there is no alpha testing at all in GLES 2

Reply:
This was specifically added in #10036 because it was only broken on *one* GPU, so that seems hard to believe.

Reply:
It was also broken in GLES 2 in each attempt of enabling shaders. Read the [OGLES2 reference](https://www.khronos.org/registry/OpenGL-Refpages/es2.0/), there are no alpha test-related functions (glAlphaFunc, GL_ALPHA_TEST in glEnable). And IrrLicht built-in shaders emulate the test too, take a look at `COGLES2TransparentAlphaChannelRef.fsh` in the IrrLicht shaders.

Reply:
Ah yea, GLES previously used Irrlicht's builtin shaders (emulating the a fixed pipeline) so of course this didn't make a difference.